### PR TITLE
update webpack typings

### PR DESCRIPTION
**Pull request by bcherny**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://webpack.js.org/configuration/stats/, https://webpack.js.org/configuration/devtool/
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: webpack/index.d.ts [hunk 1]
```diff
@@ -10,8 +10,8 @@ declare namespace webpack {
     interface Configuration {
         context?: string;
         entry?: string | string[] | Entry;
-        /** Choose a developer tool to enhance debugging. */
-        devtool?: string;
+        /** Choose a style of source mapping to enhance the debugging process. These values can affect build and rebuild speed dramatically. */
+        devtool?: 'eval' | 'inline-source-map' | 'cheap-eval-source-map' | 'cheap-source-map' | 'cheap-module-eval-source-map' | 'cheap-module-source-map' | 'eval-source-map' | 'source-map' | 'nosources-source-map' | 'hidden-source-map' | 'nosources-source-map' | boolean;
```
Comment (mattlewis92) [inline webpack/index.d.ts]: All of these can be prefixed with `@`, `#` or `#@`: https://webpack.github.io/docs/configuration.html#devtool

Maybe it's better to just string with the string type?

**Review by mattlewis92 — CHANGES_REQUESTED** [review]

File: webpack/index.d.ts (hunk 1 above)
Comment (bcherny) [inline webpack/index.d.ts]: Added them to the union. This gives better safety than `string`, at the cost of typings legibility. IMO the tradeoff is worth it - what do you think?

Comment (mattlewis92) [inline webpack/index.d.ts]: Fair enough 👍

**Review by mattlewis92 — APPROVED** [review]

**Comment by mattlewis92** [comment]
👍

**Pull request merged**